### ISSUE TITLE
How to recover from failed Try with function returning other Try?

### ISSUE THREAD

**Issue by isbjorn** [aol/cyclops-react#445]
I have some code like this:

```
Supplier<Try<Boolean, RuntimeException>> execute = ...

boolean isOk = execute.get();
```

and i want to try again and call ```recoverWith``` if the first execution failed:

```
boolean isOk = execute.get().recoverWith(ex -> execute.get());
```

but current signature of ```recoverWith``` do not allow to pass functions returning ```Try```:

```
public Success<T, X> recoverWith(Function<? super X, ? extends Success<T, X>> fn);
```

can you make it like scala version of [Try](http://www.scala-lang.org/api/2.9.3/scala/util/Try.html):

```
public Success<T, X> recoverWith(Function<? super X, ? extends Try<T, X>> fn);
```

or is there a better way to recover with function returning ```Try```?

**Comment by johnmcclean-aol** [comment]
No you are right, this is an oversight / bug. Will fix in 1.0.5

**Comment by johnmcclean-aol** [comment]
We should have fix with this in it released tomorrow, Success shouldn't be directly exposed in the API like that and is too restrictive as an input / return type for that method.

You can also use the visit method

```java
execute.get()
       .visit(Try::success,this::myMethod);

public Try<T,Throwable> myMethod(Throwable exception)
```

There are overloaded versions that accept either a Function or a Supplier for the failure case

```java
execute.get()
       .visit(Try::success,this::myMethod);

public Try<T,Throwable> myMethod()
```

**Comment by johnmcclean-aol** [comment]
This is released in 1.0.5.

The to method could also be used here. It exists to allow users to fluent inject custom operators, and although I haven't run it to check - I think you could just write 

```java
execute.get()
       .to(t->execute.get());
```
You can us the to method to  fluently inject custom operators e.g.

```java
execute.get()
       .to(myCustomOp(a->a*2));
```
